test/services/extensibility-service.ts

```diff
@@ -0,0 +1,636 @@
+		testInjector.register("fs", {});
+		testInjector.register("logger", stubs.LoggerStub);
+		testInjector.register("npm", {});
+		testInjector.register("options", {
+			profileDir: "profileDir"
+		});
+		testInjector.register("requireService", {
+			require: (pathToRequire: string): any => undefined
+		});
+		return testInjector;
+	};
+
+	describe("installExtension", () => {
+		describe("fails", () => {
+			it("when extensions dir does not exist and trying to create it fails", async () => {
+				const expectedErrorMessage = "Unable to create dir";
+				const testInjector = getTestInjector();
+				const fs: IFileSystem = testInjector.resolve("fs");
+				fs.exists = (pathToCheck: string): boolean => false;
+				fs.createDirectory = (dirToCreate: string): void => {
+					throw new Error(expectedErrorMessage);
+				};
+
+				const extensibilityService: IExtensibilityService = testInjector.resolve(ExtensibilityService);
+				await assert.isRejected(extensibilityService.installExtension("extensionToInstall"), expectedErrorMessage);
+			});
+
+			it("when extensions dir exists, but default package.json is missing and trying to create it fails", async () => {
+				const expectedErrorMessage = "Unable to write json";
+				const testInjector = getTestInjector();
+				const fs: IFileSystem = testInjector.resolve("fs");
+				fs.exists = (pathToCheck: string): boolean => path.basename(pathToCheck) !== constants.PACKAGE_JSON_FILE_NAME;
+				fs.writeJson = (pathToFile: string, content: any) => {
+					throw new Error(expectedErrorMessage);
+				};
+
+				const extensibilityService: IExtensibilityService = testInjector.resolve(ExtensibilityService);
+				await assert.isRejected(extensibilityService.installExtension("extensionToInstall"), expectedErrorMessage);
+			});
+
+			it("when npm install fails", async () => {
+				const expectedErrorMessage = "Unable to install package";
+				const testInjector = getTestInjector();
+				const fs: IFileSystem = testInjector.resolve("fs");
+				fs.exists = (pathToCheck: string): boolean => true;
+				const npm: INodePackageManager = testInjector.resolve("npm");
+				npm.install = async (packageName: string, pathToSave: string, config?: any): Promise<any> => {
+					throw new Error(expectedErrorMessage);
+				};
+
+				const extensibilityService: IExtensibilityService = testInjector.resolve(ExtensibilityService);
+				await assert.isRejected(extensibilityService.installExtension("extensionToInstall"), expectedErrorMessage);
+			});
+		});
+
+		describe("passes correct arguments to npm install", () => {
+			const getArgsPassedToNpmInstallDuringInstallExtensionCall = async (userSpecifiedValue: string, testInjector?: IInjector): Promise<any> => {
+				testInjector = testInjector || getTestInjector();
+				const fs: IFileSystem = testInjector.resolve("fs");
+				fs.exists = (pathToCheck: string): boolean => true;
+
+				fs.readDirectory = (dir: string): string[] => {
+					return [userSpecifiedValue];
```

Comment:
You could use the short-hand syntax here:
`fs.readDirectory = (dir: string) => [userSpecifiedValue]`